packages/wpcom-checkout/src/checkout-labels.ts

```diff
@@ -59,14 +63,26 @@ export function getSublabel( serverCartItem: ResponseCartProduct ): string {
 		}
 	}
 
-	if ( ! isRenewalItem && serverCartItem.months_per_bill_period === 1 ) {
-		return String( translate( 'Billed monthly' ) );
+	if ( isAddOn( serverCartItem ) && ! isRenewalItem ) {
+		return String( translate( 'Add-On' ) );
 	}
 
 	if ( isRenewalItem ) {
 		return String( translate( 'Renewal' ) );
 	}
 
+	if ( isMonthlyProduct( serverCartItem ) ) {
+		return String( translate( 'Billed monthly' ) );
+	}
+
+	if ( isYearly( serverCartItem ) ) {
+		return String( translate( 'Billed annually' ) );
+	}
+
+	if ( isBiennially( serverCartItem ) ) {
+		return String( translate( 'Billed every two years' ) );
+	}
+
 	return '';
```

Comment:
We could also consider explicitly labeling it "One-time purchase" if it's a one-time purchase, I guess?